lib/model/QubitMenu.php

```diff
@@ -52,7 +52,15 @@ public function getPath($options = array())
     $aliases = array(
       '%profile%' => sfContext::getInstance()->routing->generate(null, array('module' => 'user', 'slug' => sfContext::getInstance()->user->getUserSlug())),
       '%currentId%' => sfContext::getInstance()->request->id,
-      '%currentSlug%' => @sfContext::getInstance()->request->getAttribute('sf_route')->resource->slug
+      '%currentSlug%' => @sfContext::getInstance()->request->getAttribute('sf_route')->resource->slug,
+
+      // 'currentRealm' is used by menu items on the Institutional Block as part of the enable_institutional_scoping feature.
+      // Try using search-realm if available, else try resource->id from sf_route, else try repos param on request.
+      '%currentRealm%' => (@sfContext::getInstance()->user->getAttribute('search-realm')
```

Comment:
I'd avoid both using @ and chaining ternary operators and do this before $aliases is assigned if possible. The PHP docs warn against chaining ternary ops since there are subtleties in PHP that are non-obvious when you do it, and the @ operator... I think it's better to explicitly check things with isset() and things like that.